File: AdminArea/AdminArea.Web/Modules/ServerTypes/Productcom/ProductcomColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { ProductcomRow } from "./ProductcomRow";

export interface ProductcomColumns {
    Id: Column<ProductcomRow>;
    Name: Column<ProductcomRow>;
    Description: Column<ProductcomRow>;
    Price: Column<ProductcomRow>;
    Image: Column<ProductcomRow>;
    CategoryName: Column<ProductcomRow>;
    Date: Column<ProductcomRow>;
}

export class ProductcomColumns extends ColumnsBase<ProductcomRow> {
    static readonly columnsKey = 'Productcom.Productcom';
    static readonly Fields = fieldsProxy<ProductcomColumns>();
}

File: AdminArea/AdminArea.Web/Modules/Productcom/Productcom/ProductcomColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace AdminArea.Productcom.Columns;

[ColumnsScript("Productcom.Productcom")]
[BasedOnRow(typeof(ProductcomRow), CheckNames = true)]
public class ProductcomColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string Name { get; set; }
    public string Description { get; set; }
    public decimal Price { get; set; }
    public string Image { get; set; }
    public string CategoryName { get; set; }
    public DateTime Date { get; set; }
}